Date: Sat, 30 Sep 2006 17:25:30 +0200
From: Olaf Hering <olaf@aepfle.de>
Subject: [PATCH] AC power not detected if no battery present
Patch-mainline: yes

/proc/pmu/info contains AC Power: 0 when booting without battery.
Force AC Power, it will be updated whenever the battery state changes.

Signed-off-by: Olaf Hering <olh@suse.de>

---
 drivers/macintosh/via-pmu.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

--- linux-2.6.19.orig/drivers/macintosh/via-pmu.c
+++ linux-2.6.19/drivers/macintosh/via-pmu.c
@@ -169,7 +169,7 @@ static int option_server_mode;
 
 int pmu_battery_count;
 int pmu_cur_battery;
-unsigned int pmu_power_flags;
+unsigned int pmu_power_flags = PMU_PWR_AC_PRESENT;
 struct pmu_battery_info pmu_batteries[PMU_MAX_BATTERIES];
 static int query_batt_timer = BATTERY_POLLING_COUNT;
 static struct adb_request batt_req;
